### PR TITLE
Fixes #31500 - Add basic schedule to job wizard

### PR DESCRIPTION
**Pull request by MariaAga**
Only add components without the functionality and without the advanced options.
The Repeat field will be added more in depth in a different pr
~~Depends on: https://github.com/theforeman/foreman_remote_execution/pull/551~~

Screenshot from 2020-12-13 13-08-01

### CODE REVIEW

File: webpack/react_app/components/JobWizard/steps/Schedule/index.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+  const repeatTypes = [
+    __('Does not repeat'),
+    __('Cronline'),
+    __('Monthly'),
+    __('Weekly'),
+    __('Daily'),
+    __('Hourly'),
+  ];
+
+  const [repeatType, setSepeatType] = useState(repeatTypes[0]);
```
Comment (MariaAga) [inline webpack/react_app/components/JobWizard/steps/Schedule/index.js:26]: Type, should be `setRepeatType`

File: webpack/react_app/components/JobWizard/steps/Schedule/QueryType.js [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import { FormGroup, Radio } from '@patternfly/react-core';
+import { translate as __ } from 'foremanReact/common/I18n';
+import { helpLabel } from '../form/FormHelpers';
+
+export const QueryType = () => {
+  const [type, setType] = useState(null);
+  const types = {
+    static: 'static',
+    dynamic: 'dynamic',
+  };
```
Comment (MariaAga) [inline webpack/react_app/components/JobWizard/steps/Schedule/QueryType.js:11]: Change to isTypeStatis true/false values instead

File: webpack/react_app/components/JobWizard/steps/Schedule/ScheduleType.js [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+        id="schedule-type-now"
+        label={__('Execute now')}
+      />
+      <Radio
+        isChecked={type === types.future}
+        name="schedule-type"
+        onChange={() => setType(types.future)}
+        id="schedule-type-future"
+        label={__('Schedule for future execution')}
+      />
```
Comment (adamruzicka) [inline webpack/react_app/components/JobWizard/steps/Schedule/ScheduleType.js:26]: I'd vote for one of these being pre-selected.

File: webpack/react_app/components/JobWizard/steps/Schedule/QueryType.js [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+        id="query-type-static"
+        label={__('Static query')}
+      />
+      <Radio
+        isChecked={type === types.dynamic}
+        name="query-type"
+        onChange={() => setType(types.dynamic)}
+        id="query-type-dynamic"
+        label={__('Dynamic query')}
+      />
```
Comment (adamruzicka) [inline webpack/react_app/components/JobWizard/steps/Schedule/QueryType.js:49]: I'd vote for one of these being pre-selected.

File: webpack/react_app/components/JobWizard/steps/Schedule/index.js [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        </GridItem>
+        <GridItem span={1} />
+        <GridItem span={5}>
+          <TextInput
+            id="repeat-amount"
+            value={repeatAmount}
+            type="number"
+            onChange={setRepeatAmount}
+            placeholder={__('Repeat N times')}
+          />
```
Comment (adamruzicka) [inline webpack/react_app/components/JobWizard/steps/Schedule/index.js:51]: Only non-negative numbers make sense here, but it seems to accept anything

File: webpack/react_app/components/JobWizard/steps/Schedule/index.js [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+  const [starts, setStarts] = useState('');
+  const [ends, setEnds] = useState('');
+  return (
+    <Form>
+      <Title headingLevel="h2">{__('Schedule')}</Title>
+      <ScheduleType />
+      <Grid>
+        <GridItem span={6}>
+          <SelectField
+            fieldId="reapet-select"
```
Comment (adamruzicka) [inline webpack/react_app/components/JobWizard/steps/Schedule/index.js:37]: Typo `repeat-select`

File: webpack/react_app/components/JobWizard/steps/Schedule/index.js (hunk 5 above)
Comment (adamruzicka) [reply webpack/react_app/components/JobWizard/steps/Schedule/index.js:51]: Also this should be hidden if the `Does not repeat` is selected in the schedule dropdown

File: webpack/JobWizard/steps/Schedule/index.js [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+            validated={repeatValidated}
+          >
+            <TextInput
+              isDisabled={repeatType === repeatTypes.noRepeat}
+              id="repeat-amount"
+              value={repeatAmount}
+              type="number"
+              onChange={newValue => handleRepeatInputChange(newValue)}
+              placeholder={__('Repeat N times')}
+            />
```
Comment (MariaAga) [inline webpack/JobWizard/steps/Schedule/index.js:71]: Need to check the firefox issue here (no Onchange fired when input type is number)

File: webpack/JobWizard/steps/Schedule/StartEndDates.js [hunk 8]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from 'react';
+import { FormGroup, TextInput, Checkbox } from '@patternfly/react-core';
+import PropTypes from 'prop-types';
+import { translate as __ } from 'foremanReact/common/I18n';
+
+// TODO: change to datepicker
```
Comment (MariaAga) [inline webpack/JobWizard/steps/Schedule/StartEndDates.js:6]: https://github.com/theforeman/foreman_remote_execution/pull/567

**Comment by adamruzicka** [comment]
What would be the purpose of `Starts` and `Ends` fields if `Execute now` is selected in schedule type?

**Comment by MariaAga** [comment]
Starts: As discussed with Maria the designer, if a user will select a date the radio button will change from execute now. ( this logic is applied in the datepicker pr).
Ends: If execute now is selected, does it mean it only runs once?

**Comment by adamruzicka** [comment]
>  Ends: If execute now is selected, does it mean it only runs once?

Originally yes. Execute now meant "run it now and run it only once" and the ui didn't allow you to input a combination of execute now with end date set

File: webpack/JobWizard/steps/Schedule/index.js [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+      />
+      <StartEndDates
+        starts={starts}
+        setStarts={setStarts}
+        ends={ends}
+        setEnds={setEnds}
+      />
+      <Button variant="link" className="advanced-scheduling-button" isInline>
+        {__('Advanced scheduling')}
+      </Button>
```
Comment (laviro) [inline webpack/JobWizard/steps/Schedule/index.js:35]: What should happen when clicking on it? is there a follow-up PR?

File: webpack/JobWizard/steps/Schedule/index.js [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+import { RepeatOn } from './RepeatOn';
+import { QueryType } from './QueryType';
+import { StartEndDates } from './StartEndDates';
+import { repeatTypes } from '../../JobWizardConstants';
+
+const Schedule = () => {
+  const [repeatType, setRepeatType] = useState(repeatTypes.noRepeat);
+  const [repeatAmount, setRepeatAmount] = useState('');
+  const [starts, setStarts] = useState('');
+  const [ends, setEnds] = useState('');
```
Comment (laviro) [inline webpack/JobWizard/steps/Schedule/index.js:14]: should we handle some basic date validation here?

**Review by laviro — COMMENTED** [review]
Thanks @MariaAga, looks great!
a couple of inline comments,
and I saw that all values get cleared when moving back/forward between steps.

**Comment by MariaAga** [comment]
@laviro Please notice that this is only a basic pr. All of the issues you mentions are handled in different follow up prs.
You can see the planned prs here: https://projects.theforeman.org/issues/31178

File: webpack/JobWizard/steps/Schedule/index.js (hunk 9 above)
Comment (laviro) [reply webpack/JobWizard/steps/Schedule/index.js:35]: OK I see you have https://projects.theforeman.org/issues/32804 to track it

File: webpack/JobWizard/steps/Schedule/index.js (hunk 10 above)
Comment (laviro) [reply webpack/JobWizard/steps/Schedule/index.js:14]: is there a Redmine for this one?

Comment (laviro) [reply webpack/JobWizard/steps/Schedule/index.js:14]: maybe in the same Redmine? you can add `Fix values get cleared on scheduling tab when moving back/forward between steps`

File: webpack/JobWizard/steps/Schedule/__tests__/QueryType.test.js [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+import { QueryType } from '../QueryType';
+
+jest.spyOn(patternfly, 'FormGroup');
+patternfly.FormGroup.mockImplementation(props => <div props={props} />);
+const fixtures = {
+  renders: {},
+};
+
+describe('QueryType', () => {
+  testComponentSnapshotsWithFixtures(QueryType, fixtures);
```
Comment (laviro) [inline webpack/JobWizard/steps/Schedule/__tests__/QueryType.test.js:13]: Can you remove the snapshots testing?
we can add an integration test that just clicks on stuff and checks if API calls occurred and things like that.

**Comment by MariaAga** [comment]
Values between tabs is here: https://github.com/theforeman/foreman_remote_execution/pull/567 
validation is here: https://projects.theforeman.org/issues/32667
advanced schedule is here: https://projects.theforeman.org/issues/32804
repeat field is here: https://projects.theforeman.org/issues/32824

File: webpack/JobWizard/steps/Schedule/__tests__/QueryType.test.js (hunk 11 above)
Comment (MariaAga) [reply webpack/JobWizard/steps/Schedule/__tests__/QueryType.test.js]: Removed, will add integration testing to the next pr as in this one is only a visual pr https://github.com/theforeman/foreman_remote_execution/pull/567

File: webpack/JobWizard/steps/Schedule/__tests__/StartEndDates.test.js [hunk 12]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import * as patternfly from '@patternfly/react-core';
+import { render, fireEvent, screen } from '@testing-library/react';
+import { StartEndDates } from '../StartEndDates';
+
+jest.spyOn(patternfly, 'FormGroup');
+patternfly.FormGroup.mockImplementation(props => <div props={props} />);
```
Comment (laviro) [inline webpack/JobWizard/steps/Schedule/__tests__/StartEndDates.test.js:7]: is it still needed?

**Review by laviro — APPROVED** [review]
Thanks, nothing to add except for a small leftover (?)
waiting for the follow-ups :)

File: webpack/JobWizard/steps/Schedule/__tests__/StartEndDates.test.js (hunk 12 above)
Comment (MariaAga) [reply webpack/JobWizard/steps/Schedule/__tests__/StartEndDates.test.js]: Nope, nice catch!

**Comment by xprazak2** [comment]
Test failures do not seem relevant as this does not touch any ruby code. Any additional concerns or is this ready?

**Comment by adamruzicka** [comment]
A bit of a concern on the design side, but let's go with this for now.

**Comment by laviro** [comment]
Since we dropped snapshots the PRs are much smaller,
Would it make sense to add more logic in them in the upcoming PRs?

**Comment by adamruzicka** [comment]
Thank you @MariaAga & @laviro !

**Pull request merged**